packages/apollo/src/fetch-schema.ts

```diff
@@ -18,47 +20,74 @@ import { EndpointConfig } from "./config";
 import { engineLink, getIdFromKey } from "./engine";
 import { SCHEMA_QUERY } from "./operations/schema";
 
-const introspection = gql(introspectionQuery);
+const introspection = gql(getIntrospectionQuery());
 
-export async function fromFile(
+async function buildIntrospectionSchemaInLocalGraphQLContext(
+  source: string | Source
+): Promise<IntrospectionSchema> {
+  const schema: GraphQLSchema = buildSchema(source);
+  const executionResult: ExecutionResult = await graphql(
+    schema,
+    getIntrospectionQuery()
+  );
+
+  if (executionResult.errors) {
+    console.error(executionResult.errors);
+    throw new Error("No data received during introspection query execution.");
```

Comment:
Should this report the errors as part of the thrown `Error`?

Reply:
Yes, I'll add that.  I actually started refactoring to use a more global helper implementation of a common pattern across this whole repo:

```js
errors.map(({ message }: Error) => message).join("\n"))
```

But I stopped short due to time.  Thanks for raising this very correct suggestion.

Reply:
Fixed via 72bf1540ea244cff912c7d15919aa27ae7e9040a.